plugins/lookup/etcd3.py

```diff
@@ -0,0 +1,223 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+    author:
+    - Eric Belhomme <ebelhomme@fr.scc.com>
+    lookup: etcd3
+    short_description: Get key values from etcd3 server
+    description:
+    - Retrieves key values and/or key prefixes from etcd3 server using its native gRPC API.
+    - Try to reuse M(etcd3) options for connection parameters, but add support for some C(ETCDCTL_*) environment variables.
+    - See U(https://github.com/etcd-io/etcd/tree/master/Documentation/op-guide) for etcd overview.
+
+    options:
+        _terms:
+            description:
+            - The list of keys (or key prefixes) to look up on the etcd3 server.
+            type: list
+            elements: str
+            required: True
+        prefix:
+            description:
+            - Look for key or prefix key.
+            type: bool
+            default: False
+        endpoints:
+            description:
+            - Counterpart of C(ETCDCTL_ENDPOINTS) enviroment variable.
+              Specify the etcd3 connection with and URL form eg. C(https://hostname:2379)  or C(<host>:<port>) form
+            - mutually exclusive with I(host) and I(port)
+            env:
+            - name: ETCDCTL_ENDPOINTS
+            default: '127.0.0.1:2379'
+            type: str
+        host:
+            description:
+            - etcd3 listening client host.
+            - mutually exclusive with I(endpoints)
+            type: str
+        port:
+            description:
+            - etcd3 listening client port.
+            - mutually exclusive with I(endpoints)
+            type: int
+        ca_cert:
+            description:
+            - etcd3 CA authority.
+            env:
+            - name: ETCDCTL_CACERT
+            type: str
+        cert_cert:
+            description:
+            - etcd3 client certificate.
+            env:
+            - name: ETCDCTL_CERT
+            type: str
+        cert_key:
+            description:
+            - etcd3 client private key.
+            env:
+            - name: ETCDCTL_KEY
+            type: str
+        timeout:
+            description:
+            - Client timeout.
+            default: 60
+            env:
+            - name: ETCDCTL_DIAL_TIMEOUT
+            type: int
+        user:
+            description:
+            - Authentified user name.
+            env:
+            - name: ETCDCTL_USER
+            type: str
+        password:
+            description:
+            - Authentified user password.
+            env:
+            - name: ETCDCTL_PASSWORD
+            type: str
+
+    notes:
+    - I(endpoints) and I(host), I(port) are mutually exclusive. Using these options together will raise an error.
+    seealso:
+    - module: etcd3
+    - lookup: etcd
```

Comment:
I don't think this works. You probably have to do:
```suggestion
    - ref: etcd_lookup
```

Reply:
is there a way to test this ?

Reply:
AFAIK no.

Reply:
(I looked at the source to be sure.)